src/redis-cli.c

```diff
@@ -1192,8 +1192,15 @@ static sds cliFormatReplyJson(sds out, redisReply *r) {
     case REDIS_REPLY_MAP:
         out = sdscat(out,"{");
         for (i = 0; i < r->elements; i += 2) {
-            out = cliFormatReplyJson(out, r->element[i]);
-
+            redisReply *key = r->element[i];
+            if (key->type == REDIS_REPLY_STATUS ||
+                key->type == REDIS_REPLY_STRING ||
+                key->type == REDIS_REPLY_VERB) {
+                out = cliFormatReplyJson(out, key);
```

Comment:
looking at the code, it seems to me that always going through the `else` would yield the same results (just a bit less efficient).
please ack.

i guess we can keep the efficient way, the only downside is the risk that one of these will get updated in the outer switch-case and this code will remain stale, but since it's the same function, that's not a real risk

Reply:
As I tested with cloning #9462 like I put the description on this PR, through the `else`, the not string type will be escaped and stringified.

I tried to stringify at each switch-case, but I wonder if it's better to stringify only the map-key, it means to keep the original type of map-value.

I only came up with only solution handling it at each switch-case that adding the `stringify` argument to `cliFormatReplyJson` to control the return type.

For example.
```
cliFormatReplyJson(sds out, redisReply *r, int stringify) {
...
    case REDIS_REPLY_INTEGER:
        if (stringify) {
            // convert from Int to String
        } else {
            out = sdscatprintf(out,"%lld",r->integer);
            break;
        }
...
```
Do you have any ideas?

Reply:
If pass REDIS_REPLY_STRING to the `else` block, first `sdscatrepr` in `cliFormatReplyJson` returns the string enclosed by double quote, and `sdscatrepr` escapes and enclosed again, so the double quote added by the first call will be escaped.

```
./src/redis-cli -3 hset test3 hoge fuga
(integer) 1

./src/redis-cli -3 hgetall test3
1# "hoge" => "fuga"

./src/redis-cli -3 --json hgetall test3
{"\"hoge\"":"fuga"}
```

Reply:
ohh, didn't expect that.
well, so let's take your version.
anything else before we merge it?

Reply:
LGTM